test/FunctionalTests/Microsoft.OData.Client.Tests/Serialization/HttpClientRequestMessageTests.cs

```diff
@@ -18,5 +28,185 @@ public void UnwrapAggregateException()
             var exception = Assert.Throws<DataServiceTransportException>(() => msg.EndGetResponse(task));
             Assert.StartsWith("System.Exception: single exception\r\n", exception.Message);
         }
+
+        [Fact]
+        public void WhenHttpClientHandlerProviderIsSet_UsesReturnedHttpClientHandler_ToMakeRequest()
+        {
+            // Arrange
+            string expectedResponse = "Foo";
+            using (var handler = new MockHttpClientHandler(expectedResponse))
+            {
+
+                var httpClientHandlerProvider = new MockHttpClientHandlerProvider(handler);
+
+                var args = new DataServiceClientRequestMessageArgs(
+                    "GET",
+                    new Uri("http://localhost"),
+                    useDefaultCredentials: false,
+                    usePostTunneling: false,
+                    new Dictionary<string, string>(),
+                    httpClientHandlerProvider);
+
+                using (var request = new HttpClientRequestMessage(args))
+                {
+                    // Act
+                    IODataResponseMessage response = request.GetResponse();
+                    var stream = response.GetStream();
+                    string contents = "";

```

Comment:
```suggestion
                    string contents = "";
                    using (IODataResponseMessage response = request.GetResponse())
                    using (var stream = response.GetStream())
```

Better to be explicit about this stuff I think...

Reply:
`IODataResponseMessage` is not `IDisposable` though...